ios/FluentUI/DrawerVnext/Drawer.swift

```diff
@@ -0,0 +1,266 @@
+                    draggedOffsetWidth = value.translation.width
+                }
+            }
+            .onEnded { _ in
+                if let draggedOffsetWidth = draggedOffsetWidth {
+                    if abs(draggedOffsetWidth) < snapWidth {
+                        state.isExpanded = true
+                    } else {
+                        state.isExpanded = false
+                    }
+                    self.draggedOffsetWidth = nil
+                }
+            }
+    }
+}
+
+/**
+ `DrawerHost` is UIKit wrapper required to host UIViewController as content.
+ */
+public struct DrawerModal {
+
+    public struct DrawerShimView: View {
+
+        private var drawer: Drawer<DrawerContentViewController>
+
+        public var drawerState: DrawerState {
+            return drawer.state
+        }
+
+        init(contentViewController: UIViewController) {
+            drawer = Drawer(content: DrawerContentViewController(contentViewController: contentViewController))
+        }
+
+        public var body: some View {
+            ZStack {
+                drawer.edgesIgnoringSafeArea(.all)
+            }
+        }
+    }
+
+    public struct DrawerContentViewController: UIViewControllerRepresentable {
+
+        private var contentView: UIViewController
+
+        init(contentViewController: UIViewController) {
+            self.contentView = contentViewController
+        }
+
+        public func makeUIViewController(context: Context) -> UIViewController {
+            return contentView
+        }
+
+        public func updateUIViewController(_ uiViewController: UIViewController, context: Context) {}
+
+        public typealias UIViewControllerType = UIViewController
+    }
+}
+
+@objc(MSFDrawerVnext)
+
+/// UIKit wrapper that exposes the SwiftUI Drawer implementation
+open class DrawerVnext: UIHostingController<DrawerModal.DrawerShimView> {
+
+    @objc open var drawerState: DrawerState {
+        return self.rootView.drawerState
+    }
+
+    @objc public init(contentViewController: UIViewController) {
+        super.init(rootView: DrawerModal.DrawerShimView(contentViewController: contentViewController))
+        transitioningDelegate = self
+        drawerState.delegate = self
+        view.backgroundColor = .clear
+        modalPresentationStyle = .overFullScreen
+    }
+
+    @objc required dynamic public init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+    }
+}
+
+extension DrawerVnext: DrawerStateDelegate {
+    // two way binding if state is changed
+    func drawerDidCollapsed() {
+        dismiss(animated: true)
+    }
+}
+
+extension DrawerVnext: UIViewControllerTransitioningDelegate, UIViewControllerAnimatedTransitioning {
+    public func transitionDuration(using transitionContext: UIViewControllerContextTransitioning?) -> TimeInterval {
+        return 0
+    }
+
+    public func animateTransition(using transitionContext: UIViewControllerContextTransitioning) {
```

Comment:
I found a bug where if you drag from the right it will display the drawer from the left side anyways:

https://user-images.githubusercontent.com/68076145/102934669-71fb3a00-4459-11eb-8691-1c2660cddf17.mov

Reply:
The other bug also shown in this video is the one you mentioned before about displaying it from the left and then from the right.

Reply:
The left to right pan gesture is correct, as far the animation when switching side is concerned would be addressed as part of this [issue](https://github.com/microsoft/fluentui-apple/issues/357)